src/main/java/de/tum/in/www1/artemis/service/GroupNotificationService.java

```diff
@@ -7,220 +7,145 @@
+            return;
+        }
+        // Create and send the notification.
+        saveAndSend(GroupNotificationFactory.createNotification(attachment, userService.getUser(), GroupNotificationType.STUDENT, NotificationType.ATTACHMENT_CHANGE));
     }
 
-    private GroupNotification createAttachmentUpdatedGroupNotification(Attachment attachment, String notificationText) {
-        Course course = attachment.getLecture().getCourse();
-        String title = "Attachment " + attachment.getName() + " updated";
-        User user = userService.getUser();
-        GroupNotificationType type = GroupNotificationType.STUDENT;
-        GroupNotification groupNotification = new GroupNotification(course, title, notificationText, user, type);
-        groupNotification.setTarget(groupNotification.getAttachmentUpdated(attachment.getLecture()));
-        return groupNotification;
+    /**
+     * Notify students groups about an exercise opened for practice.
+     *
+     * @param exercise that has been opened for practice
+     */
+    public void notifyStudentGroupAboutExercisePractice(Exercise exercise) {
+        saveAndSend(GroupNotificationFactory.createNotification(exercise, userService.getUser(), GroupNotificationType.STUDENT, NotificationType.EXERCISE_PRACTICE));
     }
 
+    /**
+     * Notify student groups about an exercise started.
+     *
+     * @param exercise that has been started
+     */
     public void notifyStudentGroupAboutExerciseStart(Exercise exercise) {
-        String title = "Exercise started";
-        String notificationText = "Exercise \"" + exercise.getTitle() + "\" just started.";
-        notifyStudentGroupAboutExerciseChange(exercise, title, notificationText);
-    }
-
-    public void notifyStudentGroupAboutExerciseVisibility(Exercise exercise) {
-        String title = "New exercise available";
-        String notificationText = "Exercise \"" + exercise.getTitle() + "\" is now available.";
-        notifyStudentGroupAboutExerciseChange(exercise, title, notificationText);
-    }
-
-    public void notifyStudentGroupAboutExercisePractice(Exercise exercise) {
-        String title = "Exercise open for practice";
-        String notificationText = "Exercise \"" + exercise.getTitle() + "\" is now open for practice.";
-        notifyStudentGroupAboutExerciseChange(exercise, title, notificationText);
+        saveAndSend(GroupNotificationFactory.createNotification(exercise, userService.getUser(), GroupNotificationType.STUDENT, NotificationType.EXERCISE_STARTED));
     }
 
     /**
-     * Notify student group about an exercise update
+     * Notify student groups about an exercise update.
      *
      * @param exercise that has been updated
-     * @param notificationText that should be displayed
+     * @param notificationText that should be displayed  TODO: can be removed in the future as notification's text attribute is not used in the client
```

Comment:
Could this be removed now or are there difficulties associated with removing it?

Reply:
Forgot to work on that. Actually, we need to keep the notification text as there is an option to set a custom notification message when updating an attachment or an exercise.